drivers/hmis_csv_importer/app/jobs/hmis_csv_importer/cleanup/expire_base_job.rb

```diff
@@ -88,36 +83,38 @@ def process_model(model)
       # TODO: this can be removed (or at least the count query could be removed once we're comfortable with the results)
       log "Start Processing: #{model.table_name}, rows overall: #{model.with_deleted.count}"
 
-      expire_data(model)
+      with_tmp_table(model) do |table_name|
+        populate_tmp_table(model, table_name)
+        write_expirations(model, table_name)
+      end
 
       # TODO: this can be removed (or at least the count query could be removed once we're comfortable with the results)
       log "Completed Processing: #{model.table_name}, rows expired: #{model.with_deleted.where(expired: true).count} in #{elapsed_time(Time.current - start_time)}"
     end
 
-    private def expire_data(model)
-      tmp_table_prefix = model.table_name.downcase
-      tmp_table_name = "#{tmp_table_prefix}_tmp_exp"
-      tmp_model_name = "#{model.class.name.demodulize}Expire"
-      tmp_class = HmisCsvImporter::TempTable.create_temporary_table(table_name: tmp_table_name, model_name: tmp_model_name)
-      begin
-        populate_tmp_table(model, tmp_table_name)
-        write_expirations(model, tmp_table_name)
-      ensure
-        tmp_class.drop
+    private def with_tmp_table(model)
+      prefix = model.table_name.downcase
+      table_name = "#{prefix}_tmp_exp"
+      model.connection.create_table table_name, id: false, temporary: true do |t|
+        t.references :source, null: false, index: false
       end
+      yield(table_name)
+    ensure
+      # drop tmp table explicitly since we will continue processing additional tables with this session
+      model.connection.drop_table table_name
     end
 
     private def write_expirations(model, tmp_table_name)
       model.connection.execute(expiration_update_query(model, tmp_table_name))
     end
 
+    # this could be batched if needed
     private def expiration_update_query(model, tmp_table_name)
       <<~SQL
         UPDATE #{model.quoted_table_name} source_table
         SET expired = true
         FROM #{tmp_table_name} tmp_table
         WHERE tmp_table.source_id = source_table.id
-        AND tmp_table.batch_id < #{min_age_protected_id}
```

Comment:
Without this, we need to make sure this filter gets applied on insert/select.

Reply:
ah oops, dropped that by mistake. Thanks!